src/NServiceBus.AcceptanceTests/Core/Routing/When_sending_from_incoming_pipeline.cs

```diff
@@ -0,0 +1,117 @@
+﻿namespace NServiceBus.AcceptanceTests.Core.Routing
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using AcceptanceTesting.Customization;
+    using EndpointTemplates;
+    using NServiceBus.Pipeline;
+    using NUnit.Framework;
+
+    public class When_sending_from_incoming_pipeline : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_use_default_routing_when_empty_send_options()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<EndpointA>(e => e
+                    .When(s => s.SendLocal(new TriggerMessage())))
+                .WithEndpoint<EndpointB>()
+                .WithEndpoint<EndpointC>()
+                .Done(c => c.MessageBReceived && c.MessageCReceived)
+                .Run(TimeSpan.FromSeconds(15));
+
+            Assert.IsTrue(context.MessageBReceived);
+            Assert.IsTrue(context.MessageCReceived);
+        }
+
+        class Context : ScenarioContext
+        {
+            public bool MessageBReceived { get; set; }
+            public bool MessageCReceived { get; set; }
+        }
+
+        class EndpointA : EndpointConfigurationBuilder
+        {
+            public EndpointA()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.ConfigureTransport().Routing().RouteToEndpoint(typeof(MessageToEndpointB), typeof(EndpointB));
+                    c.ConfigureTransport().Routing().RouteToEndpoint(typeof(MessageToEndpointC), typeof(EndpointC));
+                    c.Pipeline.Register(new IncomingBehaviorWithSendLocal(), "sends a message as part of an incoming message pipeline");
+                });
+            }
+
+            class IncomingBehaviorWithSendLocal : Behavior<IIncomingLogicalMessageContext>
+            {
+                public override async Task Invoke(IIncomingLogicalMessageContext context, Func<Task> next)
+                {
+                    await context.Send(new MessageToEndpointC());
```

Comment:
I think we added it just as a safeguard while we were testing the first solutions. To detect if we mess up with this scenario. We didn't add any extra routing state in the behavior send because the other scenario (the original that uses an outgoing scenario) didn't need it, it failed just by using  SendLocal from the endpoint. Removing the test for now.